sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1182,6 +1233,78 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
       (c, evPrim, evNull) => code"$evPrim = $c != 0;"
   }
 
+  private[this] def castTimestampToIntegerCode(
```

Comment:
`Integral` is adjective. "Integral code" seems weird to me. 
We can call it `castTimestampToIntegralTypeCode` if you insist. I didn't use it because the name is a bit long.

Reply:
well, `Integer` is a specific data type, so I think this name is misleading...your suggested one is fine to me